src/System.Linq.Parallel/tests/Combinatorial/SourcesAndOperators.cs

```diff
@@ -0,0 +1,304 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Test
+{
+    public partial class ParallelQueryCombinationTests
+    {
+        private const int defaultStart = 8;
+        private const int defaultSize = 16;
+
+        private static IEnumerable<LabelledOperation> UnorderedRangeSources()
+        {
+            // The difference between this and the existing sources is I need more control over the range creation.
```

Comment:
Nit: can we avoid the use of "I" in the code?  It'd be better to use "we" or simply avoid the use of pronouns.